astropy/coordinates/sky_coordinate.py

```diff
@@ -227,7 +230,7 @@ def _parse_inputs(self, args, kwargs):
                 # coord_kwargs will contain keys like 'ra', 'dec', 'distance'
                 # along with any frame attributes like equinox or obstime which
                 # were explicitly specified in the coordinate object (i.e. non-default).
-                coord_kwargs = _parse_coordinate_arg(args[0], frame, units)
+                coord_kwargs = _parse_coordinate_arg(args[0], frame, units, kwargs)
```

Comment:
@astrofrog @eteq - I think this comment is still unresolved, though it's not a showstopper.  I've reproduced the failure that @eteq showed above so I understand a fix is not so simple, but in the current code (now in master) there is the following inconsistency:
```
In [1]: from astropy.coordinates import *

In [2]: sc = SkyCoord('1d', '2d', frame='fk5')

In [3]: sc2 = SkyCoord(sc, obstime='J2002')

In [4]: sc2.obstime
Out[4]: 'J2002'

In [5]: sc2 = SkyCoord([sc, sc], obstime='J2002')
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-5-753b93d8b73e> in <module>()
----> 1 sc2 = SkyCoord([sc, sc], obstime='J2002')

/Users/aldcroft/git/astropy/astropy/coordinates/sky_coordinate.pyc in __init__(self, *args, **kwargs)
    146         # creating the internal self._sky_coord_frame object
    147         args = list(args)  # Make it mutable
--> 148         kwargs = self._parse_inputs(args, kwargs)
    149 
    150         # Set internal versions of object state attributes

/Users/aldcroft/git/astropy/astropy/coordinates/sky_coordinate.pyc in _parse_inputs(self, args, kwargs)
    262                     raise ValueError("Coordinate attribute '{0}'={1!r} conflicts with "
    263                                      "keyword argument '{0}'={2!r}"
--> 264                                      .format(attr, coord_value, valid_kwargs[attr]))
    265                 valid_kwargs[attr] = coord_value
    266 

ValueError: Coordinate attribute 'obstime'=None conflicts with keyword argument 'obstime'='J2002'
```

In addition as I mentioned the new `init_kwargs` argument of `_parse_coordinate_arg` is essentially orphaned because the input `kwargs` is guaranteed to be `{}` all the time.  I actually changed this in the code to be `coord_kwargs = _parse_coordinate_arg(args[0], frame, units, {})` and all tests passed.